### PR TITLE
 Instructor Results page: show students in alphabetical order #3648 

### LINKED ISSUE #3648: Instructor Results page: show students in alphabetical order
Now, the students who did not responded always appear at the end of the table. They should instead be sorted alphabetically regardless of their response status.

### PR DESCRIPTION
**Pull request by rclakmal**
Fixes #3648

### CODE REVIEW

**Comment by kanghj** [comment]
@rclakmal, looks right, but I think we should first sort by team name followed by the participant's name.

**Comment by rclakmal** [comment]
@kanghj Will do the change and add a screenshot too. So if I understood it correctly, team name takes priority over the name of the giver

**Comment by rclakmal** [comment]
@kanghj @damithc Results are now sorted with giver's team first then on giver's display name

snip20160725_29

**Comment by kanghj** [comment]
@rclakmal, modifications look ok, but there's still a test failing

**Comment by rclakmal** [comment]
@kanghj Updated test pages.

**Comment by damithc** [comment]
Refactoring and more tests requested.

**Comment by rclakmal** [comment]
@damithc Added extra test to cover default search. What did you expect from refactor side ?

File: src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java [hunk 1]
```diff
@@ -962,8 +965,22 @@ private InstructorFeedbackResultsQuestionTable buildQuestionTableAndResponseRows
...
+                        return ComparisonChain.start()
+                                .compare(a1.getGiverTeam(), a2.getGiverTeam())
+                                .compare(a1.getGiverDisplayableIdentifier(),
+                                        a2.getGiverDisplayableIdentifier())
+                                .result();
+                    }
+                });
             }
             
         }
```
Comment (damithc) [inline src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java]: Note how the `if` branch is much neater than the `else` branch. Extract the sorting to a separate method (e.g. `sortAlphabetically(...)`, put it where it belongs (usually, as a static method in the class of objects being sorted) and unit test that method separately. Things like sorting are independent small functionalities that should be implemented separately and unit tested separately (i.e. don't use integration/system tests to test such small standalone functionalities).

**Comment by rclakmal** [comment]
@chowyb Refactored and added more unit tests as instructed by @damithc. Also kept the selenium test also.

**Comment by rclakmal** [comment]
Ready for review

**Comment by wkurniawan07** [comment]
@rclakmal there's one test that needs correction here, `InstructorFeedbackResultsPageUiTest#testFronEndActions`, due to conflicting interest with #5826.

**Comment by rclakmal** [comment]
@wkurniawan07 There are no merge conflicts ? or are you talking about something else ?

**Comment by wkurniawan07** [comment]
@rclakmal the problem is not merge conflict; it's a failing test at the location that I mentioned.

**Comment by rclakmal** [comment]
@wkurniawan07 Fixed by updating the test page. Thanks for the review.

**Pull request merged**